tests/scripts/all.sh

```diff
@@ -2889,25 +2889,18 @@ component_test_psa_crypto_config_reference_ecc_ffdh_no_bignum () {
 # - component_test_tfm_config()
 common_tfm_config () {
     # Enable TF-M config
-    cp configs/tfm_mbedcrypto_config_profile_medium.h "$CONFIG_H"
-    cp configs/crypto_config_profile_medium.h "$CRYPTO_CONFIG_H"
-
-    # Adjust for the fact that we're building outside the TF-M environment.
-    #
-    # TF-M has separation, our build doesn't
-    scripts/config.py unset MBEDTLS_PSA_CRYPTO_SPM
-    scripts/config.py unset MBEDTLS_PSA_CRYPTO_KEY_ID_ENCODES_OWNER
-    # TF-M provdes its own (dummy) implemenation, from their tree
-    scripts/config.py unset MBEDTLS_AES_DECRYPT_ALT
-    scripts/config.py unset MBEDTLS_AES_SETKEY_DEC_ALT
-    # We have an OS that provides entropy, use it
-    scripts/config.py unset MBEDTLS_NO_PLATFORM_ENTROPY
+    cp configs/config-tfm.h "$CONFIG_H"
+    echo "#undef MBEDTLS_PSA_CRYPTO_CONFIG_FILE" >> "$CONFIG_H"
+    cp configs/ext/crypto_config_profile_medium.h "$CRYPTO_CONFIG_H"
 
     # Other config adjustments to make the tests pass.
     # Those should probably be adopted upstream.
     #
     # - USE_PSA_CRYPTO for PK_HAVE_ECC_KEYS
     echo "#define MBEDTLS_USE_PSA_CRYPTO" >> "$CONFIG_H"
+    # PK_[PARSE/WRITE]_C used to avoid build and link errors in test_suite_pk.c
```

Comment:
This shouldn't be needed. Do we have an issue tracking this?

Reply:
> In pk_psa_sign, if we don't define PK_WRITE_C, it needs to call mbedtls_ecp_point_write_binary. However, ECP_LIGHT is disabled under p256-m. That's why I add PK_[PARSE/WRITE]_C.

This seems like a bug in our test code and we should fix this bug rather than add a dependency.

> Regarding ASN1_[PARSE/WRITE]_C found by check_config.h for pkparse/pkwrite

This, on the other hand, is a problem with library code. It makes sense that if you enable the pk module and ECDSA (built-in or PSA, it doesn't matter), then you need to enable ASN1, because the pk module uses an ASN1 representation of ECDSA signatures. Currently `check_config.h` has this dependency with `MBEDTLS_ECDSA_C`, but with `MBEDTLS_USE_PSA_CRYPTO`, the code that uses ASN1 is in `pk*.c`, so `check_config.h` should have something like
```
#if defined(MBEDTLS_PK_C) && defined(MBEDTLS_USE_PSA_CRYPTO)
#if defined(MBEDTLS_PK_CAN_ECDSA_SIGN) && !defined(MBEDTLS_ASN1_WRITE_C)
#error "MBEDTLS_PK_C with MBEDTLS_USE_PSA_CRYPTO needs MBEDTLS_ASN1_WRITE_C for ECDSA signature"
#if defined(MBEDTLS_PK_CAN_ECDSA_VERIFY) && !defined(MBEDTLS_ASN1_PARSE_C)
#error "MBEDTLS_PK_C with MBEDTLS_USE_PSA_CRYPTO needs MBEDTLS_ASN1_PARSE_C for ECDSA verification"
#endif
#endif
```

I'm not sure this is in scope here, but it's relevant for builds with the TF-M config, so I'd prefer to fix it here.